server/src/main/java/org/opensearch/rest/RestRequest.java

```diff
@@ -578,6 +579,31 @@ public static XContentType parseContentType(List<String> header) {
         throw new IllegalArgumentException("empty Content-Type header");
     }
 
+    /**
+     * The method is only used to validate whether the values of the 2 request parameters "master_timeout" and "cluster_manager_timeout" is the same value or not.
+     * If the 2 values are not the same, throw an {@link OpenSearchParseException}.
+     * @param keys Keys of the request parameters.
+     * @deprecated The method will be removed along with the request parameter "master_timeout".
+     */
+    @Deprecated
+    public void validateParamValuesAreEqual(String... keys) {
+        // Filter the non-empty values of the parameters with the key, and get the distinct values.
+        Set<String> set = new HashSet<>();
```

Comment:
Minor improvement - you don't need a `Set` to check this. You can just track the last seen value and ensure that every subsequent value matches it. This would also force a break out of the loop as soon as there is a mismatch rather than attempting to parse all values of the input keys.

Reply:
Hi @kartg Thank you so much for your careful review and your valuable opinion!

Reply:
Updated in the commit https://github.com/opensearch-project/OpenSearch/pull/2435/commits/02e68b59478fd9eff4e5c5c7f17560a702a8c1b5. I learnt the nice way to check if all the values are the same, thank you! 👍